src/config.rs

```diff
@@ -110,6 +110,7 @@ struct GlobalConfig {
     helpers: Helpers,
     #[serde(flatten)]
     packages: BTreeMap<String, Package>,
+    variables: Option<Variables>,
```

Comment:
IMO better to remove the Option and add a `#[serde(default)]` (doesn't exist = empty map), and set the type to `BTreeMap<String, toml::value::Value>` to constrain the parsing and easier usage in the other code.

Also, rename this variable from `variables` to `settings` to avoid confusion with per-package `variables`.